paws_collector.js

```diff
@@ -549,7 +549,14 @@ class PawsCollector extends AlAwsCollector {
         Promise.all(promises).then((res) => {
             return callback(null, privCollectorState, nextInvocationTimeout);
         }).catch((err) => {
-            collector.reportErrorToIngestApi(err, () => {
+            let errorObject = {};
+            if (typeof (err) === 'object') {
+                errorObject = Object.assign(errorObject, err);
+            } else {
+                let splitErrorMessage = err.split(':');
+                errorObject.errorCode = splitErrorMessage[1].slice(1, 4);
```

Comment:
Please add a test to this `else` branch

Reply:
I think we also need to check that `errorCode` can be converted to a number, like `300 < N < 600`. Otherwise it may contain some garbage.

Reply:
Please have a look if we already have an error parsing function. I suggest we either extract this code into a new function or merge it with existing function.